internal/controller/datadogagent/component_clusteragent.go

```diff
@@ -126,6 +125,7 @@ func (r *Reconciler) cleanupV2ClusterAgent(logger logr.Logger, dda *datadoghqv2a
 	clusterAgentDeployment := &appsv1.Deployment{}
 	if err := r.client.Get(context.TODO(), nsName, clusterAgentDeployment); err != nil {
 		if errors.IsNotFound(err) {
+			deleteStatusV2WithClusterAgent(newStatus)
```

Comment:
Should this also be added in component_clustercheckrunner.go for parity ?

Reply:
..added test for CLC.